src/httpServer/server.ts

```diff
@@ -248,12 +249,15 @@ async checkPortAvailability(port: number, host: string): Promise<void> {
               }
 
               let foundArg = undefined;
-              if ((ro.apiType === APITypes.GET && marg.argSource === ArgSources.DEFAULT) || marg.argSource === ArgSources.QUERY) {
+              const isQueryMethod = ro.apiType === APITypes.GET || ro.apiType === APITypes.DELETE;
+              const isBodyMethod = ro.apiType === APITypes.POST || ro.apiType === APITypes.PUT;
```

Comment:
We may as well support `PATCH` as a third body method.  That gets us all the major verbs used in APIs.